### PR TITLE
Updated Backup section

### PR DESCRIPTION
**Pull request by benarent**
A bunch of customers have been confused with our recommend backup instructions, I hope this makes things clearer.

### CODE REVIEW

**Comment by benarent** [comment]
@fspmarshall @awly PTAL: as I've incorporated some of your recent feedback and suggestions.

File: docs/4.3/admin-guide.md [hunk 1]
```diff
@@ -2259,11 +2261,58 @@ When upgrading multiple clusters:
...
 ## Backing Up Teleport
 
+When planning a backup of Teleport, it's important to know what is where and the
+importance of each component. Teleports Proxies and Nodes are stateless, and thus
+only `teleport.yaml` should be backed up.
+
+The Auth server is Teleports brains, and depending on the backend should be backup
+in a regular cadance.
+
+For example a customer running Teleport on AWS with etd have these key items of data:
```
Comment (awly) [inline docs/4.3/admin-guide.md:2271]: `s/etd/DynamoDB/`

File: docs/4.3/admin-guide.md [hunk 2]
```diff
@@ -2259,11 +2261,58 @@ When upgrading multiple clusters:
...
+|-|-|
+| Local Users ( not SSO )  | DynamoDB |
+| Certificate Authorities | DynamoDB |
+| Trusted Clusters | DynamoDB |
+| Connectors: SSO | DynamoDB / File System  |
+| RBAC | DynamoDB / File System |
+| teleport.yaml | File System |
+| teleport.service | File System |
+| license.pem | File System |
+| TLS /. | ( File System / Outside Scope )  |
```
Comment (awly) [inline docs/4.3/admin-guide.md:2283]: `TLS /.` -> `TLS key/certificate`

File: docs/4.3/admin-guide.md [hunk 3]
```diff
@@ -2259,11 +2261,58 @@ When upgrading multiple clusters:
...
+When planning a backup of Teleport, it's important to know what is where and the
+importance of each component. Teleports Proxies and Nodes are stateless, and thus
+only `teleport.yaml` should be backed up.
+
+The Auth server is Teleports brains, and depending on the backend should be backup
+in a regular cadance.
+
+For example a customer running Teleport on AWS with etd have these key items of data:
+
+| What | Where ( Example AWS Customer ) |
```
Comment (awly) [inline docs/4.3/admin-guide.md:2273]: Also mention audit log and session recordings

File: docs/4.3/admin-guide.md [hunk 4]
```diff
@@ -2259,11 +2261,58 @@ When upgrading multiple clusters:
...
+| TLS /. | ( File System / Outside Scope )  |
+
+For this customer, we would Recommend using [AWS best practices](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/BackupRestore.html) for backing up DynamoDB. If DynamoDB is used for
+the audit log, logged events have a TTL of 1 year.
+
+| Backend | Recommend Backup Strategy  |
+|-|-|
+| dir ( local filesystem )   | Copy `data_dir/storage` and use `tctl get all` to get local state. |
+| DynamoDB | [Follow AWS Guidelines for Backup & Restore](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/BackupRestore.html) |
+| etcD | [Follow etcD Guidleines for Disaster Recovery ](https://etcd.io/docs/v2/admin_guide) |
```
Comment (awly) [inline docs/4.3/admin-guide.md:2292]: nit: etcd is all lowercase

File: docs/4.3/admin-guide.md [hunk 5]
```diff
@@ -2259,11 +2261,58 @@ When upgrading multiple clusters:
...
+| teleport.service | File System |
+| license.pem | File System |
+| TLS /. | ( File System / Outside Scope )  |
+
+For this customer, we would Recommend using [AWS best practices](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/BackupRestore.html) for backing up DynamoDB. If DynamoDB is used for
+the audit log, logged events have a TTL of 1 year.
+
+| Backend | Recommend Backup Strategy  |
+|-|-|
+| dir ( local filesystem )   | Copy `data_dir/storage` and use `tctl get all` to get local state. |
```
Comment (awly) [inline docs/4.3/admin-guide.md:2290]: I'd rephrase the strategy as: "Backup `/var/lib/teleport/storage` directory and the output of `tctl get all`".
Also, if a user backs up `/var/lib/teleport`, is there any reason to do `tctl get all`?

File: docs/4.3/admin-guide.md [hunk 6]
```diff
@@ -2259,11 +2261,58 @@ When upgrading multiple clusters:
...
+| RBAC | DynamoDB / File System |
+| teleport.yaml | File System |
+| teleport.service | File System |
+| license.pem | File System |
+| TLS /. | ( File System / Outside Scope )  |
+
+For this customer, we would Recommend using [AWS best practices](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/BackupRestore.html) for backing up DynamoDB. If DynamoDB is used for
+the audit log, logged events have a TTL of 1 year.
+
+| Backend | Recommend Backup Strategy  |
```
Comment (awly) [inline docs/4.3/admin-guide.md:2288]: nit: Why Capital Case Everywhere?

File: docs/4.3/admin-guide.md [hunk 7]
```diff
@@ -2259,11 +2261,58 @@ When upgrading multiple clusters:
...
+|-|-|
+| dir ( local filesystem )   | Copy `data_dir/storage` and use `tctl get all` to get local state. |
+| DynamoDB | [Follow AWS Guidelines for Backup & Restore](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/BackupRestore.html) |
+| etcD | [Follow etcD Guidleines for Disaster Recovery ](https://etcd.io/docs/v2/admin_guide) |
+| Firestore | [Follow GCP Guidlines for Automated Backups](https://firebase.google.com/docs/database/backups) |
+
+### Config Files
+
+We would recommend you store configs as discrete files in git.
```
Comment (awly) [inline docs/4.3/admin-guide.md:2297]: Elaborate that this refers to roles, trusted clusters, local users and auth connectors. Maybe we can invent a better term for these :thinking: 
This does not refer to `teleport.yaml` itself.

Comment (awly) [inline docs/4.3/admin-guide.md:2297]: `git` -> `a version control system`

File: docs/4.3/admin-guide.md [hunk 8]
```diff
@@ -2259,11 +2261,58 @@ When upgrading multiple clusters:
...
+
+### GitOps
+
+If running Teleport at scale it's important for teams to have an automated way to
+restore Teleport. At a high level, this is our recommend approach:
```
Comment (awly) [inline docs/4.3/admin-guide.md:2302]: I wouldn't classify this as backup/restore.
This is more about applying configs by committing files to a repo.

File: docs/4.3/admin-guide.md [hunk 9]
```diff
@@ -2259,11 +2261,58 @@ When upgrading multiple clusters:
...
+
+- Persist and backup your backend
+- Share that backend among auth servers
+- Store your configs as discrete files in git
+- Have your CI run `tctl create -f *.yaml` from that git directory, without restarting teleport service / pods in k8s
+- If you lose your backend, this setup still restores the essential configs
```
Comment (awly) [inline docs/4.3/admin-guide.md:2308]: no need to include this last bullet

File: docs/4.3/admin-guide.md [hunk 10]
```diff
@@ -2259,11 +2261,58 @@ When upgrading multiple clusters:
...
+We would recommend you store configs as discrete files in git.
+
+### GitOps
+
+If running Teleport at scale it's important for teams to have an automated way to
+restore Teleport. At a high level, this is our recommend approach:
+
+- Persist and backup your backend
+- Share that backend among auth servers
+- Store your configs as discrete files in git
```
Comment (awly) [inline docs/4.3/admin-guide.md:2306]: `git` -> `VCS`

File: docs/4.3/admin-guide.md [hunk 11]
```diff
@@ -2259,11 +2261,58 @@ When upgrading multiple clusters:
...
+- Have your CI run `tctl create -f *.yaml` from that git directory, without restarting teleport service / pods in k8s
```
Comment (awly) [inline docs/4.3/admin-guide.md:2307]: skip the ", without restarting teleport service / pods in k8s" and mention that `tctl create` must run on the auth server

File: docs/4.3/admin-guide.md [hunk 12]
```diff
@@ -2274,7 +2323,7 @@ Using `tctl get all` will backup.
     - OIDC [Teleport Enterprise]
     - Roles [Teleport Enterprise]
 
-When backing up Teleport you'll need to backup up your auth server's `data_dir/storage` directly.
+When migrating backend you'll should backup up your auth server's `data_dir/storage` directly.
```
Comment (awly) [inline docs/4.3/admin-guide.md:2326]: `you'll` -> `you`

**Review by awly — COMMENTED** [review]
This PR made me realize how incoherent our backup/restore story is.
We leak too much implementation detail to the user and don't have good naming for various pieces of data we store.

It also doesn't seem to be possible to migrate audit log or session recordings between backends.

File: docs/4.3/admin-guide.md [hunk 13]
```diff
@@ -2259,11 +2261,58 @@ When upgrading multiple clusters:
 
 ## Backing Up Teleport
 
+When planning a backup of Teleport, it's important to know what is where and the
+importance of each component. Teleports Proxies and Nodes are stateless, and thus
```
Comment (travelton) [inline docs/4.3/admin-guide.md:2265]:
```suggestion
importance of each component. Teleport's Proxies and Nodes are stateless, and thus
```

File: docs/4.3/admin-guide.md [hunk 14]
```diff
@@ -2259,11 +2261,58 @@ When upgrading multiple clusters:
...
+only `teleport.yaml` should be backed up.
+
+The Auth server is Teleports brains, and depending on the backend should be backup
```
Comment (travelton) [inline docs/4.3/admin-guide.md:2268]:
```suggestion
The Auth server is Teleport's brains, and depending on the backend should be backed up
```

File: docs/4.3/admin-guide.md [hunk 15]
```diff
@@ -2259,11 +2261,58 @@ When upgrading multiple clusters:
...
+| etcD | [Follow etcD Guidleines for Disaster Recovery ](https://etcd.io/docs/v2/admin_guide) |
+| Firestore | [Follow GCP Guidlines for Automated Backups](https://firebase.google.com/docs/database/backups) |
+
+### Config Files
+
+We would recommend you store configs as discrete files in git.
+
+### GitOps
+
+If running Teleport at scale it's important for teams to have an automated way to
```
Comment (travelton) [inline docs/4.3/admin-guide.md:2304]:
```suggestion
If running Teleport at scale, it's important for teams to have an automated way to
```

File: docs/4.3/admin-guide.md [hunk 16]
```diff
@@ -2259,11 +2261,61 @@ When upgrading multiple clusters:
...
+
+## Migrating Backends.
 As of version v4.1 you can now quickly export a collection of resources from
-Teleport. This feature set works best for local and etcd, it's currently experimental
-for AWS/GCP.
+Teleport. This feature was designed to help customers migrate from local storage
+to etcd.
 
-Using `tctl get all` will backup.
+Using `tctl get all` will retrive the below items:
```
Comment (travelton) [inline docs/4.3/admin-guide.md:2318]:
```suggestion
Using `tctl get all` will retrieve the below items:
```

**Review by travelton — COMMENTED** [review]
nits

File: docs/4.3/admin-guide.md (hunk 5 above)
Comment (webvictim) [reply docs/4.3/admin-guide.md:2293]: I think the idea with `tctl get all` was that the format would be YAML and therefore backend agnostic, so theoretically you could dump out the CAs, users etc and then restore these into a different backend.

File: docs/4.3/admin-guide.md [hunk 17]
```diff
@@ -2259,11 +2261,61 @@ When upgrading multiple clusters:
...
+| license.pem | File System |
+| TLS key/certificate | ( File System / Outside Scope )  |
+| Audit log | DynamoDB  |
+| Session recordings| S3  |
+
+
+For this customer, we would recommend using [AWS best practices](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/BackupRestore.html) for backing up DynamoDB. If DynamoDB is used for
+the audit log, logged events have a TTL of 1 year.
+
+| Backend | Recommend backup strategy  |
```
Comment (webvictim) [inline docs/4.3/admin-guide.md:2291]:
```suggestion
| Backend | Recommended backup strategy  |
```

File: docs/4.3/admin-guide.md [hunk 18]
```diff
@@ -2259,11 +2261,61 @@ When upgrading multiple clusters:
...
+| Firestore | [Follow GCP Guidlines for Automated Backups](https://firebase.google.com/docs/database/backups) |
+
+### Config Files
+
+We would recommend you store configs as discrete files in git.
+
+### GitOps
+
+If running Teleport at scale it's important for teams to have an automated way to
+restore Teleport. At a high level, this is our recommend approach:
```
Comment (webvictim) [inline docs/4.3/admin-guide.md:2305]:
```suggestion
restore Teleport. At a high level, this is our recommended approach:
```

**Comment by webvictim** [comment]
@travelton TIL about code suggestions...

mind-blown

**Comment by benarent** [comment]
also TIL https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/incorporating-feedback-in-your-pull-request#applying-a-suggested-change I'll have to batch them next time.

File: docs/4.3/admin-guide.md (hunk 7 above)
Comment (benarent) [reply docs/4.3/admin-guide.md:2300]: We call them Resources. I've updated the doc

**Comment by benarent** [comment]
> It also doesn't seem to be possible to migrate audit log or session recordings between backends.

Correct, it has been requested by customers before especially to load in session recordings into a Teleport UI. I think our stance has been, we recommend you ship it off to long term storage and use `tsh play` https://github.com/gravitational/teleport/issues/1580

**Comment by webvictim** [comment]
retest this please

File: docs/4.3/admin-guide.md [hunk 19]
```diff
@@ -2219,8 +2219,10 @@ As an extra precaution you might want to backup your application prior to upgrad
     to SHA-512 based signatures. To upgrade an existing cluster, set the following in
     your teleport.yaml:
 
+    ```bash
     teleport:
       ca_signature_algo: “rsa-sha2-512”
```
Comment (webvictim) [inline docs/4.3/admin-guide.md:2224]:
```suggestion
      ca_signature_algo: "rsa-sha2-512"
```

Change the smart quotes -> regular quotes so they can be copy/pasted.

File: docs/4.3/admin-guide.md [hunk 20]
```diff
@@ -2259,11 +2261,62 @@ When upgrading multiple clusters:
 
 ## Backing Up Teleport
 
+When planning a backup of Teleport, it's important to know what is where and the
+importance of each component. Teleport's Proxies and Nodes are stateless, and thus
+only `teleport.yaml` should be backed up.
+
+The Auth server is Teleport's brains, and depending on the backend should be backed up
+in a regular cadance.
```
Comment (webvictim) [inline docs/4.3/admin-guide.md:2269]:
```suggestion
regularly.
```

(can't decide on "at a regular cadence" versus "in a regular cadence", so let's just change it to "regularly"?)

File: docs/4.3/admin-guide.md [hunk 21]
```diff
@@ -2274,7 +2327,7 @@ Using `tctl get all` will backup.
     - OIDC [Teleport Enterprise]
     - Roles [Teleport Enterprise]
 
-When backing up Teleport you'll need to backup up your auth server's `data_dir/storage` directly.
+When migrating backend you should backup up your auth server's `data_dir/storage` directly.
```
Comment (webvictim) [inline docs/4.3/admin-guide.md:2330]:
```suggestion
When migrating backends, you should back up your auth server's `data_dir/storage` directly.
```

**Review by webvictim — APPROVED** [review]
LGTM, few last nits

**Review by awly — APPROVED** [review]
